lightning/src/chain/keysinterface.rs

```diff
@@ -195,9 +195,16 @@ impl Readable for SpendableOutputDescriptor {
 // TODO: We should remove Clone by instead requesting a new ChannelKeys copy when we create
 // ChannelMonitors instead of expecting to clone the one out of the Channel into the monitors.
 pub trait ChannelKeys : Send+Clone {
-	/// Gets the commitment seed for a specific commitment number
-	/// Note that the commitment number starts at (1 << 48) - 1 and counts backwards
-	fn commitment_secret(&self, idx: u64) -> [u8; 32];
+	/// Gets the per-commitment point for a specific commitment number
+	///
+	/// Note that the commitment number starts at (1 << 48) - 1 and counts backwards.
+	fn get_per_commitment_point<T: secp256k1::Signing + secp256k1::Verification>(&self, idx: u64, secp_ctx: &Secp256k1<T>) -> PublicKey;
+	/// Gets the commitment seed for a specific commitment number, thereby revoking the commitment
+	///
+	/// After this is called, it is an error to try to sign the relevant local commitment transaction.
```

Comment:
Is it actually an error right now, or are you saying that it's a bad idea? Could potentially be clarified

Reply:
This is describing the contract for this interface method - i.e. that the caller must not try to sign this commitment.  It is up to the specific `ChannelKeys` (signer) implementation to actually enforce this, and `EnforcingChannelKeys` will do so soon.

Reply:
Ah OK -- I think RL's other docs tend to phrase "requirements that implementers of a trait must satisfy" a bit differently. To be consistent with our other docs, I'd rephrase to "After this method is called, the implementor of this trait must ensure that it will refuse sign the relevant local commitment transaction."

Reply:
rephrased, please take another look